sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/HnswVectorSearchAlgorithmConfiguration.java

```diff
@@ -91,7 +91,7 @@ public static HnswVectorSearchAlgorithmConfiguration fromJson(JsonReader jsonRea
 
                         if ("kind".equals(fieldName)) {
                             String kind = reader.getString();
-                            if (!KIND.equals(kind)) {
+                            if (!KIND.toString().equals(kind)) {
```

Comment:
The type `HnswVectorSearchAlgorithmConfiguration` is not comparable to a `String`, so this comparison always fails. I added `.toString()` manually, ideally this needs to be done via code-customization. @alzimmermsft for notice.

Reply:
This appears to be a code generator issue when the discriminator type isn't a `string`. I'll create an update to the generator so that this doesn't need to be done every time.

Reply:
@alzimmermsft the same deserialization issues appears in `AzureOpenAIVectorizer`, `CustomVectorizer`, `ExhaustiveKnnVectorSearchAlgorithmConfiguration`, `RawVectorQuery`, and `VectorizableTextQuery`